src/FSharpLint.Client/FSharpLintToolLocator.fs

```diff
@@ -0,0 +1,256 @@
+module FSharpLint.Client.FSharpLintToolLocator
+
+open System
+open System.ComponentModel
+open System.Diagnostics
+open System.IO
+open System.Text.RegularExpressions
+open System.Runtime.InteropServices
+open StreamJsonRpc
+open FSharpLint.Client.LSPFSharpLintServiceTypes
+
+let private supportedRange = SemanticVersioning.Range(">=v0.21.3") //TODO: proper version
+
+let private (|CompatibleVersion|_|) (version: string) =
+    match SemanticVersioning.Version.TryParse version with
+    | true, parsedVersion ->
+        if supportedRange.IsSatisfied(parsedVersion, includePrerelease = true) then
+            Some version
+        else
+            None
+    | _ -> None
+let [<Literal>] fsharpLintToolName = "dotnet-fsharplint"
+    
+let private (|CompatibleToolName|_|) toolName =
+    if toolName = fsharpLintToolName then
+        Some toolName
+    else
+        None
+
+let private readOutputStreamAsLines (outputStream: StreamReader) : string list =
+    let rec readLines (outputStream: StreamReader) (continuation: string list -> string list) =
+        let nextLine = outputStream.ReadLine()
+
+        if isNull nextLine then
+            continuation []
+        else
+            readLines outputStream (fun lines -> nextLine :: lines |> continuation)
+
+    readLines outputStream id
+
+let private startProcess (ps: ProcessStartInfo) : Result<Process, ProcessStartError> =
+    try
+        Ok(Process.Start ps)
+    with
+    | :? Win32Exception as win32ex ->
+        let pathEnv = Environment.GetEnvironmentVariable "PATH"
+
+        Error(
+            ProcessStartError.ExecutableFileNotFound(
+                ps.FileName,
+                ps.Arguments,
+                ps.WorkingDirectory,
+                pathEnv,
+                win32ex.Message
+            )
+        )
+    | ex -> Error(ProcessStartError.UnExpectedException(ps.FileName, ps.Arguments, ex.Message))
+
+let private runToolListCmd (Folder workingDir: Folder) (globalFlag: bool) : Result<string list, DotNetToolListError> =
+    let ps = ProcessStartInfo("dotnet")
+    ps.WorkingDirectory <- workingDir
+
+    if ps.EnvironmentVariables.ContainsKey "DOTNET_CLI_UI_LANGUAGE" then
+        ps.EnvironmentVariables.["DOTNET_CLI_UI_LANGUAGE"] <- "en-us"
```

Comment:
@MrLuje doesn't this line above work in both cases? I mean, if the key was already there and if the key was not (this way, you don't need to call ContainsKey method)

Reply:
Yep, good catch